### ISSUE TITLE
Parsing unicode regexes can cause strange behavior in Node 4

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#8020]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 4.7.3
* **npm Version:** 2.15.11

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml
parserOptions:
  ecmaVersion: 6
rules:
  eqeqeq: [error, always, {'null': never}]
```

**What did you do? Please include the actual source code causing the issue.**

```js
foo === /abc/u;
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
1:5  error  Expected '==' and instead saw '==='  eqeqeq
```

---

This is happening because the `node.value` property of the regex literal node is `null` in Node 4, because unicode regular expressions don't exist in that version. As a result, the rule assumes that the node is a `null` literal.

The rule can be updated to check the `regex` property of the literal to make sure it's not a regular expression. However, I wonder if there is a better way to represent this at the parser level -- it seems unintuitive that `node.type === 'Literal' && node.value === null` should match against anything other than a `null` literal.
